src/sql/src/session/vars.rs

```diff
@@ -821,6 +831,42 @@ impl SessionVars {
     }
 }
 
+/// If the given variable name and/or input is deprecated, return a corresponding updated value,
+/// otherwise return the original.
+///
+/// This method was introduced to gracefully handle the rename of the `mz_introspection` cluster to
+/// `mz_cluster_server`. The plan is to remove it once all users have migrated to the new name. The
+/// debug logs will be helpful for checking this in production.
+// TODO(#27285) remove this after sufficient time has passed
+fn compat_translate<'a, 'b>(name: &'a str, input: VarInput<'b>) -> (&'a str, VarInput<'b>) {
+    if name == CLUSTER.name() {
+        if let Ok(value) = CLUSTER.parse(input) {
+            if value.format() == "mz_introspection" {
+                tracing::debug!(
+                    github_27285 = true,
+                    "encountered deprecated `cluster` variable value: mz_introspection",
+                );
+                return (name, VarInput::Flat("mz_catalog_server"));
+            }
+        }
+    }
+
+    if name == "auto_route_introspection_queries" {
```

Comment:
This string is hard coded three times, could it be a `const` instead?

Reply:
Yep, done.